src/js/data/index.ts

```diff
@@ -0,0 +1,96 @@
+import { Observable, of, throwError, timer } from "rxjs";
+import { RequestResponse } from "@dcos/http-service";
+import { makeExecutableSchema } from "graphql-tools";
+
+import {
+  fetchPackageVersions,
+  PackageVersionsResponse
+} from "#SRC/js/data/CosmosClient";
+import Config from "#SRC/js/config/Config";
+import { filter, map, retry, switchMap } from "rxjs/operators";
+import { Package, PackageSchema } from "#SRC/js/types/Package";
+import { PackageVersionSchema } from "#SRC/js/types/PackageVersion";
+
+export interface ResolverArgs {
+  fetchPackageVersions: (
+    packageName: string
+  ) => Observable<RequestResponse<PackageVersionsResponse>>;
+  pollingInterval: number;
+}
+
+export interface GeneralArgs {
+  [key: string]: any;
+}
+
+export interface PackageQueryArgs {
+  name: string;
+}
+
+function isPackageQueryArgs(args: GeneralArgs): args is PackageQueryArgs {
+  return (args as PackageQueryArgs).name !== undefined;
+}
+
+export const resolvers = ({
+  fetchPackageVersions,
+  pollingInterval
+}: ResolverArgs) => ({
+  Package: {
+    versions(parent: { name: string }, _args: GeneralArgs) {
+      if (!parent.name) {
+        return throwError("Package name must be available to resolve versions");
+      }
+
+      const versions$ = timer(0, pollingInterval).pipe(
+        switchMap(() => fetchPackageVersions(parent.name).pipe(retry(2))),
+        filter(
+          ({ code }: RequestResponse<PackageVersionsResponse>) => code < 300
+        ),
+        map(
+          ({
+            response: { results }
+          }: RequestResponse<PackageVersionsResponse>) =>
+            Object.keys(results).map(version => ({
+              version,
+              revision: results[version]
+            }))
+        )
+      );
+
+      return versions$;
+    }
+  },
+  Query: {
+    package(_parent: {}, args: GeneralArgs = {}) {
+      if (!isPackageQueryArgs(args)) {
+        return throwError(
+          "Package resolver arguments aren't valid for type PackageQueryArgs"
+        );
+      }
+
+      return of({ name: args.name });
+    }
+  }
+});
+
+const baseSchema = `
+type Query {
+  package(name: String!): Package
+}
+`;
+export const schemas: string[] = [
+  PackageVersionSchema,
+  PackageSchema,
+  baseSchema
+];
+
+export interface Query {
+  package: Package | null;
+}
+
+export default makeExecutableSchema({
+  typeDefs: schemas,
+  resolvers: resolvers({
+    fetchPackageVersions,
+    pollingInterval: Config.getRefreshRate()
```

Comment:
Why would we be willing to poll the endpoint every 2 seconds?

Reply:
It wouldn't be. I initially didn't have that query using the timer at all because I wasn't sure we _should_ poll packageVersions. But then added it back for consistency with other queries. 

How should we decided where its right to return a polling observable vs not, and what the polling interval should be?

Reply:
That's a good question. While we need to answer it in general in this particular case we want to get this data once, as we visit the settings page. Otherwise I'd argue that we should rather put the polling into the rxjs client implementing a `poll` in addition to `request` and `stream`. Am I missing something?

Reply:
no, that sounds good to me. I'll remove the `timer` all together which I prefer in this case.